### ISSUE TITLE
[3.0.7] [Galacticraft] Game freezes at load screen when I first go to the moon.

### ISSUE THREAD

**Issue by dalsio** [DarkPacks/SevTech-Ages#2624]
## Bug Report
Saw this issue: #1695 and it pretty much matches my own experience, but it was closed and marked stale so I didn't know if I should add on to it or open a new one. I attempt to fly to the moon for the first time, l go to the destination selection screen, launch to the moon, and rather quickly I see my gui, character, and world background. I hear the "Stage Completed" sound but almost immediately it starts loading again and never stops, requiring termination. For reference, I'm running the game on a potato which causes longer than normal load times and I also seem to be having this issue #2299 since when I attempt to reload the game, I spawn in the middle of the sky without my rocket (but with all my items and gear, so no death) and fall into the middle of a freshly made crater (with item drops and such)  which suggests to me that the game was running in the background while loading, requiring a landing, and since I was loading crashed the rocket. I tested this twice (reloaded the save from before the freeze), but I only have the log from the second time.

Relevant Issues:
#1695 
#2299 

## Possible Solution
It may be a bad interaction with poor performance (as I said, I'm running on a potato). The dimension load, terrain generation, stage advancement, crash, explosion, and death may all be happening right on top of each other or in the wrong order, causing some errors. I don't know how feasible it is, but perhaps stopping the game/rocket temporarily while loading, giving more time to land, running the stage advancement earlier like when you launch the rocket or delaying the stage advancement until you "safely" land all might be possible solutions.

On my end, I might be able to work around it by holding space while loading as was suggested in #2299, or I might be able to alt-tab as soon as I launch to the moon to force the game to pause as soon as it's able to maybe give my PC some time to catch up.

## Steps to Reproduce
1. Have not so stellar performance.
1. Gain full space gear (suit, oxygen gear, oxygen tanks), fully fueled t1 max storage rocket, launch pad, and many items in inventory. (My space suit crafted with full energy. Is that supposed to happen?)
2. Launch the rocket. I tested it both at night in the rain and during the day.
3. Once in the selection screen, select the moon and launch.
4. Take quite a second to load and then freeze.

## Logs
* Client/Server Log: https://gist.github.com/dalsio/09063af7df3f8a27edb5c4af745f3324

## Client Information
* Modpack Version: 3.0.7
* Java Version: 1.8.0.171
* Launcher Used: Twitch, modified
* Memory Allocated: 4gb
* Server/LAN/Single Player: Single
* Optifine Installed: No
* Shaders Enabled: No

**Comment by SereneRebel** [comment]
also confirm hang. i have also experienced very long load times entering dimensions for the first time but the moon has hung for over 10 mins. minecraft didnt go "not responding" so i figured it wasnt some small infinite loop. while stick no inputs work no menus i had to close minecraft all together 

once the game restarted i fell to my death and exploded. i had the chute and all preparations

**Comment by justinrusso** [comment]
It's hard to tell if the issue is related to the same performance issue or if there is more to this. Our update to GameStages 2.0 will improve the performance when switching dimensions, but the moon taking as long as it does compare to others makes it stand out as something that has more to it.

**Comment by iceTburg** [comment]
I am having the same issue when going to the moon via rocket.  I am having (I think) client crashes a lot of the time when I change dimensions, which kick me out of the world as if the server crashed.  Minecraft itself does not crash, and I usually just have to reload my saved game a couple of times.  However when traveling to the moon it completely hangs and I have to exit Minecraft and relaunch.  Happy to provide more information, but am not getting a crash log, so not sure what to provide.

**Comment by dalsio** [comment]
Update: I just came back to the Overworld after spending some time on the moon and experienced an identical infinite loading screen (obviously, there was no stage advancement).
Log: https://gist.github.com/dalsio/0a798ef78199539ac5b7e7c2bf6a85cf

I also now remember a crash quite some time ago when I came back from the hunting dimension:

I had barricaded the entrance and exit to the portal (built a small structure around it out of dirt) since I didn't know if things could come back through it or not. When I came back one time I got an ILS very similar to this one. After loading back in, I found myself dead, having spawned inside of what appeared to be solid rock. Once I respawned, I tried to follow my death compass to my grave but it said I was at y=0 at a strange spot in the world and no matter how far I dug I couldn't seem to find my grave stone (I find that my spawn position changes to some very random places every now and then if I die very quickly in rapid succession).

I noticed the last time I tried to go to the moon and crashed, when I reloaded into the game it spawned me on a very strange platform that seemed to be at around y=300 made of a bunch of colored blocks (kind of like the overworld image in the sky but with more colors). Once I jumped, I simply fell through and when attempting to use my elite reusable ender pearl to get back up, I couldn't find it again. I don't have an image of it, but I did keep a save from before launching so I might be able to use that to get a picture if you like.

So, stage advancement doesn't seem to be affecting it for me, but what does seem to matter is whether I "die" while the dimension is still loading. It may rather be an interaction with death, like with grave stones or death waypoints being created as the dimension is still loading.
